src/App.tsx

```diff
@@ -1,26 +1,313 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateCompletedTodo,
+} from './api/todos';
+import { ToDoHeader } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { ErrorType } from './types/ErrorType';
+import { Errors } from './components/Error/Error';
 
-const USER_ID = 0;
+type TempTodo = {
+  id: number;
+  completed: boolean;
+  title: string;
+  userId: number;
+};
 
 export const App: React.FC = () => {
+  const [tasks, setTasks] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [status, setStatus] = useState<Status>(Status.all);
+  const [taskTitle, setTaskTitle] = useState('');
+  const [allDone, setAllDone] = useState(false);
```

Comment:
This should be computed value. Just calculate if `tasks.length === completedTodos.length`